zerver/lib/streams.py

```diff
@@ -167,7 +167,10 @@ def can_access_stream_history_by_name(user_profile: UserProfile, stream_name: st
     can actually see this stream.
     """
     try:
-        stream = get_stream(stream_name, user_profile.realm)
+        if isinstance(operand, str):
+            stream = get_stream(operand, user_profile.realm)
+        else:
+            stream = get_stream_by_id_in_realm(operand, user_profile.realm)
     except Stream.DoesNotExist:
         return False
```

Comment:
I think I'd prefer a cleaner refactor, where we have a `access_stream_history_common` that takes a `Stream` object, and two callpoints, `can_access_stream_history_by_name` and `can_access_stream_history_by_id`.

Reply:
I lost you at the "two callpoints" part here, I am pretty sure though that you meant: we have a function, which I named as `can_access_stream_history` that takes in `Stream` object. And then we have two different `can_access_stream_by_name` and `can_access_stream_by_id` that retrieves stream and pass to `can_access_stream_history`. That is what I currently have which we can change if that's not what you expected.

Reply:
What you did is what I had in mind!